lib/Extension/LanguageServerPhpCsFixer/LanguageServerPhpCsFixerExtension.php

```diff
@@ -17,20 +22,56 @@ class LanguageServerPhpCsFixerExtension implements OptionalExtension
 
     public function load(ContainerBuilder $container): void
     {
+        $container->register(
+            PhpCsFixerProcess::class,
+            function (Container $container) {
+                $path = $container->get(FilePathResolverExtension::SERVICE_FILE_PATH_RESOLVER)->resolve($container->getParameter(self::PARAM_PHP_CS_FIXER_BIN));
+
+                return new PhpCsFixerProcess(
+                    $path,
+                    LoggingExtension::channelLogger($container, 'php-cs-fixer'),
+                    $container->getParameter(self::PARAM_ENV),
+                );
+            }
+        );
+
         $container->register(PhpCsFixerFormatter::class, function (Container $container) {
-            $path = $container->get(FilePathResolverExtension::SERVICE_FILE_PATH_RESOLVER)->resolve($container->getParameter(self::PARAM_PHP_CS_FIXER_BIN));
-            return new PhpCsFixerFormatter($path, $container->getParameter(self::PARAM_ENV));
+            return new PhpCsFixerFormatter($container->get(PhpCsFixerProcess::class));
         }, [
             LanguageServerExtension::TAG_FORMATTER => []
         ]);
+
+        $container->register(PhpCsFixerDiagnosticsProvider::class, function (Container $container) {
+            return new PhpCsFixerDiagnosticsProvider(
+                $container->get(PhpCsFixerProcess::class),
+                LoggingExtension::channelLogger($container, 'php-cs-fixer')
+            );
+        }, [
+            LanguageServerExtension::TAG_DIAGNOSTICS_PROVIDER => [
+                'name' => 'php-cs-fixer'
+            ],
+            LanguageServerExtension::TAG_CODE_ACTION_PROVIDER => []
+        ]);
+
+        $container->register(FormatCommand::class, function (Container $container) {
+            return new FormatCommand(
+                $container->get(PhpCsFixerProcess::class),
+                $container->get(ClientApi::class),
+                $container->get(LanguageServerExtension::SERVICE_SESSION_WORKSPACE),
+                LoggingExtension::channelLogger($container, 'php-cs-fixer')
+            );
+        }, [
+            LanguageServerExtension::TAG_COMMAND => [
+                'name' => 'php_cs_fixer.fix'
+            ],
+        ]);
     }
 
     public function configure(Resolver $schema): void
     {
         $schema->setDefaults([
             self::PARAM_PHP_CS_FIXER_BIN => '%project_root%/vendor/bin/php-cs-fixer',
             self::PARAM_ENV => [],
```

Comment:
maybe an option to disable diagnostics? I would prefer them to be disabled but still have the option to format the code.

Reply:
Added in one of the commits